### PR TITLE
Support pushing to the helm repo server which enable the https with tls

### PR DESCRIPTION
**Pull request by steven-zou**
**Support pushing to the helm repo server which enable the https with tls**

- add --ca-file option
- add --cert-file option
- add --key-file option
- add --insecure option
- add UT cases for the newly appended options

### CODE REVIEW

**Comment by steven-zou** [comment]
@jdolitsky 

  Please help to review the PR. Thanks a lot!

**Comment by steven-zou** [comment]
@jdolitsky 

Could you please help to review the PR? Thanks!

File: cmd/helmpush/main.go [hunk 1]
```diff
@@ -18,20 +14,29 @@ import (
...
+		password           string
+		accessToken        string
+		authHeader         string
+		contextPath        string
+		forceUpload        bool
+		useHTTP            bool
+		caFile             string
+		certFile           string
+		keyFile            string
+		InsecureSkipVerify bool
```
Comment (jdolitsky) [inline cmd/helmpush/main.go]: Can we switch this to lowercase `insecureSkipVerify` to match the other fields?

File: cmd/helmpush/main.go [hunk 2]
```diff
@@ -88,6 +93,11 @@ func newPushCmd(args []string) *cobra.Command {
 	f.StringVarP(&p.accessToken, "access-token", "", "", "Send token in Authorization header [$HELM_REPO_ACCESS_TOKEN]")
 	f.StringVarP(&p.authHeader, "auth-header", "", "", "Alternative header to use for token auth [$HELM_REPO_AUTH_HEADER]")
 	f.StringVarP(&p.contextPath, "context-path", "", "", "ChartMuseum context path [$HELM_REPO_CONTEXT_PATH]")
+	//Appended for supporting https with certificates
```
Comment (jdolitsky) [inline cmd/helmpush/main.go]: can remove this comment

File: cmd/helmpush/main.go [hunk 3]
```diff
@@ -113,6 +123,20 @@ func (p *pushCmd) setFieldsFromEnv() {
 		p.useHTTP, _ = strconv.ParseBool(v)
 	}
 
+	//Appended for supporting https with certificates
```
Comment (jdolitsky) [inline cmd/helmpush/main.go]: can remove this comment

**Comment by steven-zou** [comment]
Code refactored, please review. Thanks! @jdolitsky

**Comment by steven-zou** [comment]
@jdolitsky 

  We're using chartmuseum as our backend storage server for helm charts. We'll provide other features like RBAC and namespace isolation etc. features on top of the chartmuesum.

**Comment by steven-zou** [comment]
@jdolitsky 

  I'm also planning to enhance this push plugin to support uploading prov files together with charts at the same time (Of course, a single prov file can also be uploaded by the plugin). and --namespace/--project to push charts/prov files to the specified namespace/project when the user just configured a single root entry point of the chart repo.
e.g: herm repo add myrepo https://hostname/chartrepo
       helm push --namespace project1 mychart-0.1.0.tgz myrepo (equals to https://hostname/chartrepo/project1/mychart-0.1.0.tgz)
       helm push --namespace project2 mychart-0.1.0.tgz myrepo (equals to https://hostname/chartrepo/project2/mychart-0.1.0.tgz)

  What do you think?

**Comment by jdolitsky** [comment]
@steven-zou thanks again, change has been merged!

As for your other comments, they sound like good ideas.  Please open new issues so we can discuss semantics. #8 is also asking about provenance file support.

**Comment by jdolitsky** [comment]
@steven-zou this feature is now in 0.6.0 release

**Pull request merged**